### PR TITLE
Handle nested JAR files in getClassNamesFromPackage and reduce cognit…

### PR DESCRIPTION
**Pull request by lobocode**
This commit updates the getClassNamesFromPackage method to properly handle nested JAR files within fat JARs. It checks for a second exclamation mark in the URL to identify nested JARs and uses a JarInputStream to read the entries and load the appropriate classes. This change ensures correct class loading when the application is packaged as a fat JAR with nested dependencies and breaks down the original getClassNamesFromPackage method into smaller, more manageable functions, following SonarLint rule java:S3776. The changes improve code readability and maintainability by reducing cognitive complexity.

This is a bug fix reported in https://github.com/kubernetes-client/java/issues/1659

This pull request includes a series of changes to address the issues and suggestions pointed out by the SonarLint analysis (only critical and major smells). Each change has been committed with an appropriate commit. These changes include improvements such as code simplification, use of Optional instead of null, more informative Javadoc tags, among others.

The refactoring of the findApiResourceByGroupVersionKind method involved creating a new HashMap to store API resources by their kind and group. This new map allows for faster lookup of resources during the method's execution, improving the overall performance. The previous implementation required iterating through all resources until a match was found, resulting in a slower search. The new implementation leverages the HashMap data structure, which provides faster access to resources by key. This refactoring improves the code's maintainability, readability, and efficiency.

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/util/ModelMapper.java [hunk 1]
```diff
@@ -19,21 +19,18 @@
...
 import java.net.URLDecoder;
 import java.time.Duration;
-import java.util.ArrayList;
-import java.util.Enumeration;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/ModelMapper.java:28]: We have a style policy to disallow wildcard imports.

**Comment by brendandburns** [comment]
there's a bunch of style/whitespace changes in this PR, can you run `mvn spotless:apply` and `mvn spotless:check` and then I can give it a more detailed review.

**Comment by lobocode** [comment]
> there's a bunch of style/whitespace changes in this PR, can you run `mvn spotless:apply` and `mvn spotless:check` and then I can give it a more detailed review.

Ah, thanks for the tip (I can be a lil' scatterbrained sometimes). I already fixed it buddy !

**Comment by brendandburns** [comment]
/lgtm
/approve

It would be great to have a fatjar test at somepoint but that can be a later PR.

**Pull request merged**